cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -488,6 +475,52 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
             };
         }
     });
+    AccessEditor = AbstractEditor.extend({
+        templateName: 'access-editor',
+        className: 'edit-settings-access',
+        events : {
+            'change #prereq': 'handlePrereqSelect',
+            'keyup #prereq_min_score': 'validate'
+        },
+        afterRender: function () {
+            AbstractEditor.prototype.afterRender.call(this);
+            var prereq = this.model.get('prereq') || '';
+            var prereq_min_score = this.model.get('prereq_min_score') || '';
+            this.$('#is_prereq').prop('checked', this.model.get('is_prereq'));
+            this.$('#prereq option[value="' + prereq + '"]').prop('selected', true);
+            this.$('#prereq_min_score').val(prereq_min_score);
+            this.$('#prereq_min_score_input').toggle(prereq.length > 0);
+        },
+        handlePrereqSelect: function () {
+            var showPrereqInput = this.$('#prereq option:selected').val().length > 0;
+            this.$('#prereq_min_score_input').toggle(showPrereqInput);
+        },
+        validate: function () {
+            var minScore = this.$('#prereq_min_score').val();
+            var minScoreInt = parseInt(minScore);
+            // minScore needs to be an integer between 0 and 100
+            if (
+                minScore &&
+                (
+                    typeof(minScoreInt) === 'undefined' ||
+                    String(minScoreInt) !== minScore ||
```

Comment:
Maybe we should trim the spaces as well before the check.
parseInt would accept "40 "(a space after 40) as valid, but since "40" and "40 " are not equal, this would make the number invalid. Especially use when someone's copy-pasting and copies an additional space character, and has no clue where the error lies. Moreover, since I am a regex fanboy, I'd do this validation with a regex :-).